src/Engine/ProtoCore/Lang/Replication/Replicator.cs

```diff
@@ -214,33 +214,38 @@ public static List<ReplicationInstruction> ReductionToInstructions(List<int> red
                 {
                     //This should generally be a collection, so we need to do a one phase unboxing
                     var targets = reducedParams.Select(r => r[index]).ToList();
-                    foreach (var target in targets)
+                    StackValue target = basicList[index];
+
+                    if (!target.IsArray)
                     {
-                        if (!target.IsArray)
-                        {
-                            System.Console.WriteLine("WARNING: Replication unbox requested on Singleton. Trap: 437AD20D-9422-40A3-BFFD-DA4BAD7F3E5F");
-                            continue;
-                        }
+                        System.Console.WriteLine("WARNING: Replication unbox requested on Singleton. Trap: 437AD20D-9422-40A3-BFFD-DA4BAD7F3E5F");
+                        continue;
+                    }
 
-                        var array = runtimeCore.Heap.ToHeapObject<DSArray>(target);
-                        if (array.Count == 0)
-                        {
-                            continue;
-                        }
+                    var array = runtimeCore.Heap.ToHeapObject<DSArray>(target);
+                    if (array.Count == 0)
+                    {
+                        continue;
+                    }
 
-                        var arrayStats = ArrayUtils.GetTypeExamplesForLayer(target, runtimeCore).Values;
+                    var arrayStats = new HashSet<StackValue>();
 
-                        List<List<StackValue>> clonedList = new List<List<StackValue>>(reducedParams);
-                        reducedParams.Clear();
+                    foreach (var targetTemp in targets)
+                    {
+                        List<StackValue> temp = ArrayUtils.GetTypeExamplesForLayer(targetTemp, runtimeCore).Values.ToList();
+                        arrayStats.UnionWith(temp);
```

Comment:
what is `temp` here - I dont see it in the changes? is it somewhere else near these changes?